src/west/main.py

```diff
@@ -503,7 +503,7 @@ def main(argv=None):
     except WestUpdated:
         # West has been automatically updated. Restart ourselves to run the
         # latest version, with the same arguments that we were given.
-        os.execv(sys.executable, [sys.executable] + argv)
+        os.execv(sys.executable, [sys.executable] + [sys.argv[0]] + argv)
```

Comment:
why not use `sys.argv` directly?

Reply:
I think it has to do with wrapped commands, where main() could have been called with a set of argv != sys.argv.
Line 494:495 show that argv = sys.argv if when main(argv == None)

An example where argv != sys.argv is in post-init, where the wrapped command will has different parameters than the origonal west command.

And I didn't wanted to change that.
However, we still need sys.argv[0] in order to be able to reinvoke west.

Reply:
Oh I see  yes, I didn't realize that `argv` could be different from `sys.argv[1:]`